ui/components/app/alert-system/multiple-alert-modal/multiple-alert-modal.tsx

```diff
@@ -144,32 +146,46 @@ export function MultipleAlertModal({
   onFinalAcknowledgeClick,
   ownerId,
 }: MultipleAlertModalProps) {
-  const { alerts } = useAlerts(ownerId);
+  const { isAlertConfirmed, fieldAlerts: alerts } = useAlerts(ownerId);
 
   const [selectedIndex, setSelectedIndex] = useState(
-    alerts.findIndex((alert) => alert.key === alertKey),
+    alerts.findIndex((alert: Alert) => alert.key === alertKey),
   );
 
   const selectedAlert = alerts[selectedIndex];
+  const hasUnconfirmedAlerts = alerts.some(
+    (alert: Alert) =>
+      !isAlertConfirmed(alert.key) && alert.severity === Severity.Danger,
+  );
```

Comment:
It might be useful to move methods like above to `useAlerts` so they can be re-used.

Reply:
100%, I should have done that before. I didn't push it because I saw this improvement in your PR.